src/factory-daemon.js

```diff
@@ -127,10 +125,28 @@ class FactoryDaemon {
 
     series([
       (cb) => options.init
-        ? node.init(options.initOptions, cb)
+        ? node.init(options.initOptions, (err) => {
+          if (err) {
+            console.log(err)
+            // TODO if init fails, check if it was because the
+            // repo was already was inited, if yes, continue
+            cb(err)
+          } else {
+            cb()
+          }
+        })
         : cb(null, node),
       (cb) => options.start
-        ? node.start(options.args, cb)
+        ? node.start(options.args, (err) => {
+          if (err) {
+            console.log(err)
+            // TODO if start fails, check if it was because there
+            // is a daemon already running and connect to it instead
```

Comment:
Would appreciate some brainstorming on this one. Would the best just to move this func out https://github.com/ipfs/js-ipfsd-ctl/blob/master/src/ipfsd-daemon.js#L229-L234 and if the daemon is running, catch the addr from the `api` file in the repo?

@hugomrdias?